### DISCUSSION TITLE
Version 2.0.33 incorrectly considers Mapped[Annotated[list, int]] as list[list]

### DISCUSSION THREAD

**Discussion by Klikini** [sqlalchemy/sqlalchemy#11830] [Usage Questions] [answered]
Hi, my application has code that looks like this:

```python
# common file

from typing import Annotated

from sqlalchemy import Integer
from sqlalchemy.dialects.postgresql import ARRAY
from sqlalchemy.orm import MappedAsDataclass, DeclarativeBase

IntArray = Annotated[list, int]

class Base(MappedAsDataclass, DeclarativeBase):
    type_annotation_map = {
        IntArray: ARRAY(Integer),
    }
```

```python
# implementation file

from sqlalchemy.orm import mapped_column, Mapped

from .common import Base, IntArray

class MyModel(Base):
    my_column: Mapped[IntArray] = mapped_column(default_factory=list)
```

This worked fine with version 2.0.32, but fails with the following error after upgrading to version 2.0.33: `sqlalchemy.exc.ArgumentError: Could not locate SQLAlchemy Core type for Python type list[list] inside the 'my_column' attribute Mapped annotation`

Is this a bug in the new release, or was I doing something wrong that didn't cause problems until now?

**Comment by zzzeek** [comment, marked as answer]
regression, #11831 fix will be released tomorrow

**Reply by Klikini** [reply]
That was fast! Thank you :)

**Reply by zzzeek** [reply]
note this works with `from __future__ import annotations` which you should be using in any case so I would use that for now

**Reply by zzzeek** [reply]
also you should apply a type to that list, which also works:

```
Annotated[list[int], int]
```

the use of `list` by itself is incomplete

**Reply by Klikini** [reply]
Ah, I think I misunderstood what `Annotated` was doing in [Customizing the Type Map](https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#customizing-the-type-map).

Replacing `IntArray = Annotated[list, int]` with `IntArray = list[int]` seems to fix it as well (running with 2.0.33 _or_ 2.0.34). Am I losing anything by doing that instead of `Annotated[list[int], int]`?

I think it would be good to include an example of `list[x]` → `ARRAY(X)` type annotation mapping in the documentation, which I'm happy to contribute once I figure out exactly what it's supposed to look like.

**Reply by zzzeek** [reply]
if you want all `list[int]` to be `IntArray` globally, then you dont need to use `Annotated`.   `Annotated` is only if you want to have a single type have multiple "identities"

**Reply by Klikini** [reply]
That makes sense, thank you! My goal was to avoid redundantly specifying

`my_column: list[int] = mapped_column(ARRAY(Integer), default_factory=list)`

(and the associated import statements) every time and instead just have

`my_column: Mapped[IntArray] = mapped_column(default_factory=list)`.
